apps/authz/kinds/v0alpha1/rolespec.cue

```diff
@@ -0,0 +1,23 @@
+package v0alpha1
+
+RoleSpec: {
+	#Permission: {
+		// RBAC action (e.g: "dashbaords:read")
+		action: string
+		// RBAC scope (e.g: "dashboards:uid:dash1")
+		scope: string
+	}
+	
+	name: string
```

Comment:
Do we need this? the machine-readable name should be able to be the k8s name

Reply:
Good catch, agree with Dan.

Reply:
I had understood that the k8s name would replace the UID.

I'm trying to think about the value of the current role name (e.g. `fixed:users:writer`) I think it is currently used to distinguish between role types (`fixed`, `basic`, `managed`, ...)
Given Roles are only going to be user defined roles (`custom`), fixed and plugin roles are going to be `CoreRoles`, managed roles are going to be `ResourcePermissions`, I'm tempted to get rid of that field altogether.

Reply:
@gamab k8s name is just a string – as long as it's unique within the api group (and namespace for namespaced resources) it can be anything.

I would expect that for core roles this would always be human-readable (i.e. `users:writer`, `dashboards:admin`, etc.) and for custom roles it'd be up to the users to either define the name (e.g. when using Terraform / CLI / other as-code clients), or let the API take care of generating one (e.g. when using the UI), in which case it'd be a generic UID.

i.e. I don't think we need an extra `name` field in the spec. `title` however is fine and makes sense for a human-readable name that could be changed without re-creating the resource (since k8s names cannot be changed).

Reply:
Yes I totally agree.